File: src/main/java/jp/dataforms/fw/app/menu/page/SideMenuForm.java
package jp.dataforms.fw.app.menu.page;

import jp.dataforms.fw.menu.Menu;
import jp.dataforms.fw.menu.SideMenu;

/**
 * サイドメニューフォームクラス。
 *
 */
public class SideMenuForm extends MenuForm {
	/**
	 * コンストラクタ。
	 */
	public SideMenuForm() {
		super(null);
	}

	@Override
	protected Menu newMenuComponent() {
		SideMenu menu = new SideMenu();
		return menu;
	}


}

File: src/main/java/META-INF/resources/dataforms/app/menu/page/SideMenuForm.js
/**
 * メニューフォーム.
 */

'use strict';

import { MenuForm } from './MenuForm.js';

export class SideMenuForm extends MenuForm {
	/**
	 * ページの各エレメントとの対応付け.
	 */
	attach() {
		super.attach();
		this.menu = this.newInstance(this.sideMenu);
		this.menu.init();
		this.menu.attach();
	}
}


